### PR TITLE
Gh pages

### PR DESCRIPTION
**Pull request by darrenrector**

### CODE REVIEW

**Comment by darrenrector** [comment]
changes

File: js/script.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    var data = null;
+
+    var localStorageKey = "destinationsData";
+    var targetArea = document.getElementById("locations");
+    var ajax = document.getElementById("ajax");
+    var load = document.getElementById("load");
+    var save = document.getElementById("save");
+    var clear = document.getElementById("clear");
+
+    function firstApp(destinations) {
```
Comment (mbMosman) [inline js/script.js]: You want to make a variable/object to hold your methods.

File: js/script.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        if (typeof (localStorage) === 'undefined') {
+            targetArea.innerHTML = "Sorry, local storage is not supported for this browser.";
+        } else {
+            targetArea.innerHTML = "Loading Data...";
+            var text = localStorage.getItem(localStorageKey);
+            if (text === null) {
+                targetArea.innerHTML = "Sorry, no local data found.";
+            } else {
+                data = JSON.parse(text);
+                text(data);
```
Comment (mbMosman) [inline js/script.js]: I don't see a method called text, so I'm not sure what you are trying to do with 'text(data)'.  You want to call the method to build your HTML here.

File: js/script.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            if (i === 0) {
+                targetArea.innerHTML = "";
+            }
+
+            list = destinations[i];
+            targetArea.appendChild(firstApp(locations));
+        }
+    }
+
+    function loadLocalData() {
```
Comment (mbMosman) [inline js/script.js]: These should also be methods in your application object.

File: js/script.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        var destinations = data.locations;
+        var i, list;
+
+        for (i = 0; i < locations.length; i++) {
+            if (i === 0) {
+                targetArea.innerHTML = "";
+            }
+
+            list = destinations[i];
+            targetArea.appendChild(firstApp(locations));
```
Comment (mbMosman) [inline js/script.js]: I'm not sure what you are doing here... `firstApp` is your method above, but it looks like maybe this builds HTML for one "destination"?  If so, then maybe what you want to pass into it is `list` which isn't a list but is your first item from data.locations.

**Comment by mbMosman** [comment]
Hope this helps...

**Pull request closed without merging**